options/options.go

```diff
@@ -452,6 +454,61 @@ func (opts *ToolOptions) ParseArgs(args []string) ([]string, error) {
 	return args, err
 }
 
+// LogSensitiveOptionWarnings logs a warning for any sensitive information (i.e. passwords)
+// that appear on the command line for the --password, --uri and --sslPEMKeyPassword options.
+// This also applies to a connection string that appears as a positional argument.
+func LogSensitiveOptionWarnings(args []string) {
+	passwordMsg := "WARNING: On some systems, a password provided directly using " +
+		"--password may be visible to system status programs such as `ps` that may be " +
+		"invoked by other users. Consider omitting the password to provide it via stdin, " +
+		"or using the --config option to specify a configuration file with the password."
+
+	uriMsg := "WARNING: On some systems, a password provided directly using " +
+		"--uri may be visible to system status programs such as `ps` that may be " +
+		"invoked by other users. Consider omitting the password to provide it via stdin, " +
+		"or using the --config option to specify a configuration file with the password."
+
+	sslMsg := "WARNING: On some systems, a password provided directly using --sslPEMKeyPassword " +
+		"may be visible to system status programs such as `ps` that may be invoked by other users. " +
+		"Consider using the --config option to specify a configuration file with the password."
+
+	for i, arg := range args {
+		// Option has the form --password= but is not followed by the empty string.
+		if strings.HasPrefix(arg, "--password=") && len(arg) > 11 && arg[11:] != "" {
+			log.Logvf(log.Always, passwordMsg)
+			continue
+		}
+
+		// Option has the form --password and the next argument is not the empty string.
+		if arg == "--password" && i+1 < len(args) && args[i+1] != "" {
```

Comment:
Should this be `i+1 <= len(args)`?

Reply:
Hmm, since this condition is checking if something comes after `--password`, I think it's okay as-is.

If we have the args list `["--username", "user1", "--password", "abc123"]` then `"--password"` has an index of 2. So when we're looping and `i=2`, we'll have `i+1 = 3 < 4 = len(args)`, which shows that there's a value following `--password`.

We'd get an out-of-bounds error if our condition was instead `i+1 <= len(args)` and the args list was `["--username", "user1", "--password"]`. For `i=2` you'll get `i+1 = 3 <= 3 = len(args)` and then try to access `args[3]` which is OOB.

Btw having `--password` at the end isn't valid anyway since you'll get `error parsing command line options: expected argument for flag -p, --password`. So I thought it'd be best to not print anything and let the error occur later on.

Reply:
FWIW, it seems to me that this whole issue is moot if we just use the config values after they've been parsed into structs.